### PR TITLE
PopoverEducational: new component

### PR DESCRIPTION
**Pull request by AlbertCarreras**
#### What changed?

Popover had a eduction variant. Forking PopoverEducation to let Popover be pure building-block component and remove props from it.

Screenshot 2023-02-25 at 4 08 19 PM

WARNING: The examples will look better once we use the new editor

#### Why?

Enable future rework of Popover

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-5234)

### Checklist

- [x] Added unit and Flow Tests
- [X] Added documentation + accessibility tests
- [X] Verified accessibility: keyboard & screen reader interaction
- [X] Checked dark mode, responsiveness, and right-to-left support
- [X] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

File: docs/docs-components/COMPONENT_DATA.js [hunk 1]
```diff
@@ -1776,6 +1776,55 @@ const GENERAL_COMPONENT_LIST: $ReadOnlyArray<ListItemType> = [
       figma: null,
     },
   },
+  {
+    svg: <Popover />,
+    name: 'PopoverEducational',
+    description:
+      'PopoverEducational is a floating container that introduces users to elements on the screen. Used for education or onboarding experiences. ',
```
Comment (dangerismycat) [inline docs/docs-components/COMPONENT_DATA.js:1783]: nit: there's an extra whitespace at the end of the string

File: docs/pages/web/popovereducational.js [hunk 2]
```diff
@@ -0,0 +1,628 @@
...
+import PageHeader from '../../docs-components/PageHeader.js';
+import QualityChecklist from '../../docs-components/QualityChecklist.js';
+
+export default function DocsPage({ generatedDocGen }: {| generatedDocGen: DocGen |}): Node {
+  return (
+    <Page title={generatedDocGen?.displayName}>
+      <PageHeader
+        name={generatedDocGen?.displayName}
+        description={generatedDocGen?.description}
+        defaultCode={`
```
Comment (dangerismycat) [inline docs/pages/web/popovereducational.js:17]: Screen Shot 2023-02-27 at 2 21 41 PM

This seems a bit odd without a target element for it to be pointing to. Can we add a button or something here?

File: docs/pages/web/popovereducational.js [hunk 3]
```diff
@@ -0,0 +1,628 @@
...
+      <GeneratedPropTable generatedDocGen={generatedDocGen} />
+
+      <MainSection name="Usage guidelines">
+        <MainSection.Subsection columns={2}>
+          <MainSection.Card
+            cardSize="md"
+            type="do"
+            title="When to use"
+            description={`
+- Bringing attention to specific user interface elements for educational  or onboarding purposes.
```
Comment (dangerismycat) [inline docs/pages/web/popovereducational.js:51]: nit: I think there's some extra whitespace around `or`

File: docs/pages/web/popovereducational.js [hunk 4]
```diff
@@ -0,0 +1,628 @@
...
+          primaryAction={{ text: "Next"}}
+        />
+      )}
+    </Box>
+)}`}
+          />
+          <MainSection.Card
+            cardSize="md"
+            type="don't"
+            description="Show more than one PopoverEducational at a time. If used for onboarding, show a next button instead, to launch the next popover."
```
Comment (dangerismycat) [inline docs/pages/web/popovereducational.js:123]: Given this guidance, should we add a second popover and make the button functional?

File: docs/pages/web/popovereducational.js [hunk 5]
```diff
@@ -0,0 +1,628 @@
...
+)}`}
+          />
+        </MainSection.Subsection>
+      </MainSection>
+
+      <AccessibilitySection name={generatedDocGen?.displayName}>
+        <MainSection.Subsection
+          title="Keyboard navigation"
+          description={`
+PopoverEducational doesn't behave like regular popovers where they are open/closed upon user interaction, p.e. Tooltip, Dropdown, or ComboBox. PopoverEducational visibility is not directly controlled by the user; instead, its visibility is defined as part of a broader user experience and the user interaction engagement with this experience.
```
Comment (dangerismycat) [inline docs/pages/web/popovereducational.js:293]: typo: "_i_. e."

File: docs/pages/web/popovereducational.js [hunk 6]
```diff
@@ -0,0 +1,628 @@
...
+      )}
+    </Box>
+)}`}
+          />
+        </MainSection.Subsection>
+        <MainSection.Subsection
+          title="Custom content"
+          description={`For more flexibility, PopoverEducational allows passing children. If passed, \`message\` and \`primaryAction\` are not rendered. PopoverEducational doesn't overwrite style in children or set any padding or margin, therefore, make sure any Text's \`color\` is "light" and any Button's \`color\` is "white".
+          \`
+PENDING      `}
```
Comment (dangerismycat) [inline docs/pages/web/popovereducational.js:451]: remove this?

File: docs/pages/web/popovereducational.js [hunk 7]
```diff
@@ -0,0 +1,628 @@
...
+  );
+}
+`}
+          />
+        </MainSection.Subsection>
+      </MainSection>
+
+      <QualityChecklist component={generatedDocGen?.displayName} />
+
+      <MainSection name="Related">
```
Comment (dangerismycat) [inline docs/pages/web/popovereducational.js:610]: Seems like Popover should be here as well?

File: packages/gestalt/src/PopoverEducational.js [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+        color="blue"
+        id={id}
+        idealDirection={idealDirection}
+        onDismiss={onDismiss}
+        positionRelativeToAnchor
+        showCaret
+        role={primaryAction && !children ? 'dialog' : role}
+        size={size}
+      >
+        {message && !children ? (
```
Comment (dangerismycat) [inline packages/gestalt/src/PopoverEducational.js:172]: This logic seems a bit disconnected. I'd recommend something like
```js
{children ?? message ? (markup) : null}
```
That makes the relationship between `children` and `message` more clear

**Review by dangerismycat — APPROVED** [review]

File: docs/pages/web/popovereducational.js (hunk 5 above)
Comment (AlbertCarreras) [reply docs/pages/web/popovereducational.js:293]: p.e. is very me... cant help it

File: docs/docs-components/COMPONENT_DATA.js (hunk 1 above)
Comment (AlbertCarreras) [reply docs/docs-components/COMPONENT_DATA.js:1783]: DONE

File: docs/pages/web/popovereducational.js (hunk 2 above)
Comment (AlbertCarreras) [reply docs/pages/web/popovereducational.js:17]: Sure, I just wanted to highlight the component itself, as it's not a "paired component"

DONE

File: docs/pages/web/popovereducational.js (hunk 3 above)
Comment (AlbertCarreras) [reply docs/pages/web/popovereducational.js:51]: DONE

File: docs/pages/web/popovereducational.js (hunk 4 above)
Comment (AlbertCarreras) [reply docs/pages/web/popovereducational.js:123]: DONE

File: docs/pages/web/popovereducational.js (hunk 5 above)
Comment (AlbertCarreras) [reply docs/pages/web/popovereducational.js:293]: DONE

File: docs/pages/web/popovereducational.js (hunk 6 above)
Comment (AlbertCarreras) [reply docs/pages/web/popovereducational.js]: DONE

File: docs/pages/web/popovereducational.js (hunk 7 above)
Comment (AlbertCarreras) [reply docs/pages/web/popovereducational.js:686]: POpover is goig to become a building block, so I'd keep it separated from here.

File: packages/gestalt/src/PopoverEducational.js (hunk 8 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/PopoverEducational.js]: DONE, good catch

File: docs/pages/web/popovereducational.js (hunk 5 above)
Comment (dangerismycat) [reply docs/pages/web/popovereducational.js]: Oh, is "p.e." a thing people use? I don't think I've come across it before.

Also technically this should be "e.g.", not "i.e."…though not what they actually stand for, I remember usage like this:
`"i.e."` - "in other words"
`"e.g."` - "example(s) given"

**Pull request merged**